src/utils.cpp

```diff
@@ -94,6 +101,11 @@ void CacheableWriter::serializehandle(handle field)
     dest.append((char*)&field, sizeof(field));
 }
 
+void CacheableWriter::serializenodehandle(handle field)
+{
+    dest.append((const char*)&field, MegaClient::NODEHANDLE);
```

Comment:
I'm not sure about this one -  I think an alternative is to add functions for those specific cases when needed, so `serializechathandle()` etc, and we could make chat handles (and others) an explicit separate type too, to avoid accidental use of one in the context of another, and also to make doubly sure we write the correct entry to the database now, and enable the compiler to alert us to any mismatch between the type used and the type written to the database when making changes.   Does that sound good?  thanks

Reply:
That approach would be even better, yes. There have been small issues in the past due by misusing wrong size. However, my concern is that redefining the datatype for chat_link_handles, contact_link_handles and node_handles would require too many changes here and there. Something that requires time and that I'd prefer to not be part of this PR.
By now, I think we can leave it as your current code in this branch. And perhaps improve all that in a future ;)